core/modules/mod.rs

```diff
@@ -669,11 +663,64 @@ impl Stream for RecursiveModuleLoad {
   }
 }
 
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
-#[repr(u32)]
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+#[repr(u8)]
 pub(crate) enum AssertedModuleType {
+  /// JavaScript or WASM.
   JavaScriptOrWasm,
+  /// JSON.
   Json,
+
+  // IMPORTANT: If you add any additional enum values here, you must update `to_v8`` below!
+  /// Non-well-known module type.
+  Other(Cow<'static, str>),
+}
+
+impl AssertedModuleType {
+  pub fn to_v8<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+  ) -> v8::Local<'s, v8::Value> {
+    match self {
+      AssertedModuleType::JavaScriptOrWasm => v8::Integer::new(scope, 0).into(),
+      AssertedModuleType::Json => v8::Integer::new(scope, 1).into(),
+      AssertedModuleType::Other(ty) => {
+        v8::String::new(scope, ty).unwrap().into()
+      }
+    }
+  }
```

Comment:
Where do we even use this method? Ideally we could remove it all together (in another PR)

Reply:
It's in the snapshot serialization -- if we move to serde_v8 we can get rid of this at that point. :)